### PR TITLE
`sgkit` ancestral allele

### LINKED ISSUE #764: sgkit: Use ancestral_allele_index
Along with docs on how to create from both `AA` field in VCF and FASTA.

### PR DESCRIPTION
**Pull request by benjeffery**
Fixes #764 
Stacked on #791

### CODE REVIEW

File: tsinfer/formats.py [hunk 1]
```diff
@@ -2311,40 +2321,63 @@ def sites_metadata_schema(self):
...
-        return self.data["variant_allele"]
+        return self.data["variant_allele"][:][self.sites_mask]
+
+    @property
+    def sites_mask(self):
+        try:
+            # Often xarray will save a bool array as int8, so we need to cast,
+            # but check that a mistake hasn't been made by checking
+            # that the values are either 0 or 1
+            if da.max(self.data["variant_mask"].astype(np.int8)).compute() > 1:
```
Comment (jeromekelleher) [inline tsinfer/formats.py:2349]: int8s can be less than 0 too..

**Review by jeromekelleher — COMMENTED** [review]
Looks good

File: tsinfer/formats.py (hunk 1 above)
Comment (benjeffery) [reply tsinfer/formats.py:2385]: Fixed!

**Comment by jeromekelleher** [comment]
Will look once #791 is in (should be in directly, right?)

**Comment by benjeffery** [comment]
@jeromekelleher I've added some more tests and the warning for missing ancestral states.

**Review by jeromekelleher — APPROVED** [review]

**Pull request merged**